tslearn/docs/examples/classification/plot_shapelet_locations.py

```diff
@@ -61,18 +65,28 @@
 distances = shp_clf.transform(X_train)
 predicted_locations = shp_clf.locate(X_train)
 
-plt.figure()
-plt.title("Example locations of shapelet matches "
-          "({} shapelets extracted)".format(n_shapelets))
+f, ax = plt.subplots(2, 1, sharex=True)
 
-# Plot the test timeseries with the best matches with the shapelets
+# Plot the shapelet and align it on the best matched time series. The optimizer
+# will often enlarge the shapelet to create a larger gap between the distances
+# of both classes. We therefore normalize the shapelet again before plotting.
 test_ts_id = numpy.argmin(numpy.sum(distances, axis=1))
-plt.plot(X_train[test_ts_id].ravel())
-
-# Plot the shapelets on their best-matching locations
-for idx_shp, shp in enumerate(shp_clf.shapelets_):
-    t0 = predicted_locations[test_ts_id, idx_shp]
-    plt.plot(numpy.arange(t0, t0 + len(shp)), shp, linewidth=2)
+shap = shp_clf.shapelets_[0]
+shap = TimeSeriesScalerMinMax().fit_transform(shap.reshape(1, -1, 1)).flatten()
+pos = predicted_locations[test_ts_id, 0]
+ax[0].plot(X_train[test_ts_id].ravel())
+ax[0].plot(numpy.arange(pos, pos + len(shap)), shap, linewidth=2)
+ax[0].axvline(pos, color='k', linestyle='--', alpha=0.25)
+ax[0].set_title("The aligned extracted shapelet")
+
+# We calculate the distances from the shapelet to the timeseries ourselves.
+distances = []
+time_series = X_train[test_ts_id].ravel()
+for i in range(len(time_series) - len(shap)):
+    distances.append(numpy.linalg.norm(time_series[i:i+len(shap)] - shap))
+ax[1].plot(distances)
+ax[1].axvline(numpy.argmin(distances), color='k', linestyle='--', alpha=0.25)
```

Comment:
Maybe you could use pos here, no ?

Reply:
Yes, I did do this deliberately to "confirm" that the returned locations are really the argmins, but I can definitely change this back to `pos`! I am currently working on a "distance" example as well with the distance-transformed timeseries & decision boundary.

Am I correct that I will get the corresponding weights for the linear decision boundaries in `shp_clf.model_.get_weights()[-1]`? This gives me 4 parameters for 2 shapelets (bias term and intercept?).

Reply:
Scratch that last bit. It depends on the number of classes of course. Would you prefer a binary example with 1 decision boundary or multiclass with K decision boundaries?

Reply:
For `pos`, I understand your point, choose the option you prefer.

In order to visualize decision boundaries, I'd rather use something like:
https://nbviewer.jupyter.org/github/rtavenar/rtavenar.github.io/blob/master/teaching/ml/notebooks/01_SVM.ipynb

Of course, there is no need for a margin here.

Reply:
I just made something like this:

![Screenshot from 2020-06-07 14-56-08](https://user-images.githubusercontent.com/5750603/83969219-0c859880-a8cf-11ea-8910-066c1e977b8d.png)

Is that OK? It's building...

Reply:
I have pushed suggested changes. It's building. I think it's a great illustration both for the user guide and for the gallery!

Reply:
I made some extra additions on top of your changes. I think it would be nice to also display the timeseries as well (to better show how those markers in the distance space correspond to those time series), but perhaps it is a bit too cluttered now with all timeseries?